modules/network/src/test/java/org/apache/ignite/network/internal/AllTypesMessageGenerator.java

```diff
@@ -0,0 +1,256 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.ignite.lang.IgniteUuidGenerator;
+import org.apache.ignite.plugin.extensions.communication.MessageCollectionItemType;
+
+/**
+ * Generator for an {@link AllTypesMessage}.
+ */
+public class AllTypesMessageGenerator {
+    /**
+     * Generate a new {@link AllTypesMessage}.
+     *
+     * @param seed Random seed.
+     * @param nestedMsg {@code true} if nested messages should be generated, {@code false} otherwise.
+     * @return Message.
+     * @throws Exception If failed.
+     */
+    public static AllTypesMessage generate(long seed, boolean nestedMsg) {
+        try {
+            var random = new Random(seed);
+
+            var message = new AllTypesMessage();
+
+            Field[] fields = AllTypesMessage.class.getDeclaredFields();
+
+            for (Field field : fields) {
+                field.setAccessible(true);
+
+                TestFieldType annotation = field.getAnnotation(TestFieldType.class);
+
+                if (annotation != null) {
+                    field.set(message, randomValue(random, annotation.value(), nestedMsg));
+                }
+            }
+
+            if (nestedMsg) {
+                Field objectArrayField = AllTypesMessage.class.getDeclaredField("v");
+                objectArrayField.setAccessible(true);
+
+                Field collectionField = AllTypesMessage.class.getDeclaredField("w");
+                collectionField.setAccessible(true);
+
+                Field mapField = AllTypesMessage.class.getDeclaredField("x");
+                mapField.setAccessible(true);
+
+                Object[] array = IntStream.range(0, 10).mapToObj(unused -> generate(seed, false)).toArray();
+
+                objectArrayField.set(message, array);
+
+                List<Object> collection = IntStream.range(0, 10)
+                    .mapToObj(unused -> generate(seed, false))
+                    .collect(Collectors.toList());
+
+                collectionField.set(message, collection);
+
+                Map<String, Object> map = IntStream.range(0, 10)
+                    .boxed()
+                    .collect(Collectors.toMap(String::valueOf, unused -> generate(seed, false)));
+
+                mapField.set(message, map);
+            }
+
+            return message;
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Generate random value.
+     *
+     * @param random Seeded random.
+     * @param type Value type.
+     * @param nestedMsg {@code true} if nested messages should be generated, {@code false} otherwise.
+     * @return Random value.
+     * @throws Exception If failed.
+     */
+    private static Object randomValue(Random random, MessageCollectionItemType type, boolean nestedMsg) throws Exception {
+        Object value = null;
```

Comment:
This method will be much shorter if you remove this variable and use `return` inside branches instead